test/Sema/extension_retroactive_conformances.swift

```diff
@@ -125,10 +134,20 @@ extension Sample7: SampleProtocol1 & SampleProtocol2 {}
 
 extension Sample8: @retroactive SampleProtocol1 & SampleProtocol2 {}  // ok
 
+// FIXME: Module qualification should suppress this warning
+extension Sample8a: Library.SampleProtocol1 & Library.SampleProtocol2 {}  // ok
+// expected-warning@-1 {{extension declares a conformance of imported type 'Sample8a' to imported protocols 'SampleProtocol1', 'SampleProtocol2'; this will not behave correctly if the owners of 'Library' introduce this conformance in the future}}
+// expected-note@-2 {{add '@retroactive' to silence this warning}}
+
 extension Sample9: SampleProtocol3<Int> {}
 // expected-warning@-1 {{extension declares a conformance of imported type 'Sample9' to imported protocol 'SampleProtocol3'; this will not behave correctly if the owners of 'Library' introduce this conformance in the future}}
 // expected-note@-2 {{add '@retroactive' to silence this warning}}
 
-extension Sample10: @retroactive SampleProtocol3<Int> {}
+extension Sample9a: @retroactive SampleProtocol3<Int> {}
 
-#endif
\ No newline at end of file
+// FIXME: Module qualification should suppress this warning
+extension Sample9b: Library.SampleProtocol3<Int> {}
```

Comment:
i didn't look into the history too much here, but the [proposal](https://github.com/swiftlang/swift-evolution/blob/80f2fa54d6d938456a1720f98b4651b85a8f6e16/proposals/0364-retroactive-conformance-warning.md#source-compatibility) currently says this on suppression via qualification (emphasis mine):

> a client can silence this warning by fully-qualifying **_all_** types in the extension

is the protocol type actually the only one that requires full qualification to suppress the warning?